marathon-interface/src/main/scala/mesosphere/marathon/interface/auth/Authenticator.scala

```diff
@@ -0,0 +1,35 @@
+package mesosphere.marathon.interface.auth
+
+import mesosphere.marathon.interface.http.{ HttpResponse, HttpRequest }
+import mesosphere.marathon.interface.plugin.Plugin
+
+import scala.concurrent.Future
+
+/**
+  * Base trait for all authenticator implementations.
+  * Every authenticator can get data from an HTTP request and derive an Identity.
+  * The Identity usually is maintained in a separate Identity and Access Management System (e.g. LDAP)
+  *
+  * The Identity is passed to the [[Authorizer]].
+  * It should hold all information to make decisions possible for authorization.
+  *
+  * The Resources will make sure, that the [[Authenticator#authenticate]]
```

Comment:
This sentence is incomplete.

Did you want to say that the resources have to make sure that Authenticator#authenticate is called only once per request?